src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -63,7 +75,18 @@ static void UCS_CLASS_DELETE_FUNC_NAME(uct_cuda_ipc_iface_t)(uct_iface_t*);
 ucs_status_t uct_cuda_ipc_iface_get_device_address(uct_iface_t *tl_iface,
                                                    uct_device_addr_t *addr)
 {
-    *(uint64_t*)addr = ucs_get_system_id();
+    uct_cuda_ipc_device_addr_t *dev_addr = (uct_cuda_ipc_device_addr_t*)addr;
+    uct_cuda_ipc_iface_t *iface          = ucs_derived_of(tl_iface,
+                                                          uct_cuda_ipc_iface_t);
+    uct_cuda_ipc_md_t *md                = ucs_derived_of(iface->super.super.md,
+                                                           uct_cuda_ipc_md_t);
+
+    if (md->enable_mnnvl) {
+        dev_addr->flags = UCT_CUDA_IPC_DEVICE_ADDR_FLAG_MNNVL;
```

Comment:
maybe instead of extending device address with flags, we can set system_uuid to 0 (or some other constant) when have MNNVL? then no need to extend UCT API

Reply:
in this case we will not be able to compare system_uuids when on one node we have different GPUs with and without MNNVL support (probably unrealistic use-case)

Reply:
another case is wire compatibility. If different processes on the same node would use different UCX libs, they will not use cuda-ipc

Reply:
I don't think supporting mixed versions is so important for MNNVL at this point

Reply:
ok, but found another use-case:
we can't check whether it is the same process or not (in which case we do not have to use cuda-ipc)
If pids (iface addresses) are the same, we have to check if system ids are also the same, which is not possible with the proposed approach